SleepNumberController_Driver.groovy

```diff
@@ -139,6 +143,88 @@ def updated() {
   poll()
 }
 
+def uninstalled() {
+   deleteChildDevice(getChildDNI("outlet"))
+   deleteChildDevice(getChildDNI("underbedlight"))
+}
+
+def manageChildren() {
+    switch (state.type) {
+        case "presence":
+            def outlet = getOutlet()
+            if (!outlet) {
+                outlet = addChildDevice('hubitat','Generic Component Switch', getChildDNI("outlet"),
+                                        [ label: getChildName("Outlet"),
+                                         componentName: "outlet",
+                                         componentLabel: getChildName("Outlet"),                	                  
+                                         completedSetup:true])
+                log.info("Created Outlet child device")
+            }
+            def underbed = getUnderbed()
+            if (!underbed) {
+                outlet = addChildDevice('hubitat','Generic Component Switch', getChildDNI("underbedlight"),
+                                        [ label: getChildName("Underbed Light"),
+                                         componentName: "underbedlight",
+                                         componentLabel: getChildName("Underbed Light"),                	                  
+                                         completedSetup:true])
+                log.info("Created Underbed light child device")
+            }
+            break
+    }
+}
+
+def getOutlet() {
+    return childDevices.find({it.deviceNetworkId == getChildDNI("outlet")})
```

Comment:
Couldn't you fetch this by name (w/o the iterator) by using getChildDevice(DNI)?

Reply:
yes, that probably would be cleaner:

```suggestion
    return getChildDevice(getChildDNI("outlet"))
```